File: PayrollServer.Domain/Entities/Employee.cs
using System;
using System.Collections.Generic;

namespace PayrollServer.Domain.Entities
{
    public class Employee : BaseEntity
    {
        public Employee()
        {
            SalaryRecords = new HashSet<SalaryRecord>();
            AbsenceRecords = new HashSet<AbsenceRecord>();
            Incentives = new HashSet<Incentive>();
        }
        
        // Basic information
        public string EmployeeId { get; set; } = string.Empty; // Employee ID (not the primary key)
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public DateTime DateOfBirth { get; set; }
        public string Address { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        
        // Employment details
        public DateTime HiringDate { get; set; }
        public string EmploymentStatus { get; set; } = "Active"; // Active, Inactive, Terminated, etc.
        
        // Foreign keys
        public int DepartmentId { get; set; }
        public int JobGradeId { get; set; }
        
        // Navigation properties
        public virtual Department Department { get; set; } = null!;
        public virtual JobGrade JobGrade { get; set; } = null!;
        public virtual ICollection<SalaryRecord> SalaryRecords { get; set; }
        public virtual ICollection<AbsenceRecord> AbsenceRecords { get; set; }
        public virtual ICollection<Incentive> Incentives { get; set; }
    }
} 

File: frontend/payroll-web-app/src/types/employee.ts
// Employee types based on backend API structure

export interface Employee {
  id: number;                    // Primary key (integer)
  employeeNumber?: string;       // The display employee ID from backend EmployeeNumber
  firstName: string;
  lastName: string;
  fullName?: string;             // Computed property from backend
  email: string;
  phoneNumber?: string;
  address?: string;
  dateOfBirth: string;          // ISO date string
  hiringDate: string;           // Required - ISO date string (maps to backend HiringDate)
  employmentStatus: string;     // Required - Active/Inactive etc  
  departmentId: number;
  departmentName?: string;
  departmentIncentivePercentage?: number;
  jobGradeId: number;
  jobGradeName?: string;
  jobGradeMinSalary?: number;
  jobGradeMaxSalary?: number;
  currentSalary?: number;
  salaryEffectiveDate?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateEmployeeRequest {
  employeeId: string;             // Maps to EmployeeNumber in backend
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  address?: string;
  dateOfBirth: string;           // Will be converted to DateTime by backend
  hireDate: string;              // Will be converted to HiringDate by backend  
  departmentId: number;
  jobGradeId: number;
  initialSalary?: number;        // Optional initial salary
}

export interface UpdateEmployeeRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  address?: string;
  dateOfBirth: string;           // Will be converted to DateTime by backend
  departmentId: number;
  jobGradeId: number;
  employmentStatus: string;      // Status can be updated but not employeeId or hireDate
}

export interface EmployeesResponse {
  items: Employee[];              // Backend returns "Items", not "employees"
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

export interface EmployeeQueryParams {
  page?: number;                  // Backend expects "page", not "pageNumber"
  pageSize?: number;
  searchTerm?: string;
  departmentId?: number;
  jobGradeId?: number;
  // Note: Backend doesn't have isActive filter, uses employmentStatus
}

// Form data structure for React Hook Form
export interface EmployeeFormData {
  employeeId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  address: string;
  dateOfBirth: string;
  hireDate: string;
  departmentId: string;
  jobGradeId: string;
}

// Department and Job Grade types for dropdowns
export interface Department {
  id: number;
  name: string;
  incentivePercentage?: number;    // Matches backend DepartmentDto
  incentiveSetDate?: string;
  employeeCount: number;
  createdAt: string;
  updatedAt?: string;
}

export interface JobGrade {
  id: number;
  name: string;
  description: string;             // Required in backend
  minSalary: number;
  maxSalary: number;
  employeeCount: number;
  createdAt: string;
  updatedAt?: string;
} 